sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/jdbc/core/resultset/EncryptResultSetMetaData.java

```diff
@@ -126,7 +126,7 @@ public String getColumnName(final int column) throws SQLException {
     
     private String getLogicColumn(final String actualColumn) throws SQLException {
         for (Entry<String, String> entry : logicAndActualColumns.entrySet()) {
-            if (entry.getValue().contains(actualColumn)) {
+            if (entry.getValue().equals(actualColumn)) {
```

Comment:
you should add a test case for this corner case to make sure that this modification works well as we expect.

Reply:
@xiyelife You could add unit test in `EncryptStatementTest`.

Reply:
and look over this modification, you can use map.values().contains instead of for statement.

Reply:
> and look over this modification, you can use map.values().contains instead of for statement.
Here the code want to get the matched value's key, so must use the for each.

Reply:
> @xiyelife You could add unit test in `EncryptStatementTest`.

Ok  I will.

Reply:
> @xiyelife You could add unit test in `EncryptStatementTest`.

OK, I will.